pkg/config/generate.go

```diff
@@ -101,6 +102,8 @@ func Generate(cs *acsapi.ContainerService, c *Config) (err error) {
 	c.Version = versionLatest
 	c.TunnelHostname = strings.Replace(cs.Properties.OrchestratorProfile.OpenShiftConfig.PublicHostname, "openshift", "openshift-tunnel", 1)
 
+	// TODO: Need unique name, potentially derivative from PublicHostname
+	c.RouterLBCName = fmt.Sprintf("router-%s", strings.Split(cs.Properties.OrchestratorProfile.OpenShiftConfig.PublicHostname, ".")[1])
```

Comment:
We need unique name here for `name.eastus.cloudapp.azure.com` . I imagine as past of PP or GA this should be MS responsibility extend their plugin or make zone configurable via annotations for cases like OSA. Or we can generate some short unique string like `router-[random]-osa.eastus.cloudapp.azure.com` based on the time of creation and some unique metadata (?)

Reply:
Issue for this to track?

Reply:
I think this is fine for now.